src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -912,14 +920,46 @@ private void checkAndPlace()
         }
         else if (FreeMode.SUPPLYCAMP == Settings.instance.getFreeMode())
         {
+            final List<PlacementError> placementErrorList = new ArrayList<PlacementError>();
             if (ItemSupplyCampDeployer.canCampBePlaced(Minecraft.getMinecraft().world, Settings.instance.getPosition(),
-                    Settings.instance.getActiveStructure().getSize(BlockUtils.getRotation(Settings.instance.getRotation()))))
+                    Settings.instance.getActiveStructure().getSize(BlockUtils.getRotation(Settings.instance.getRotation())), placementErrorList))
             {
                 pasteNice();
             }
             else
             {
-                LanguageHandler.sendPlayerMessage(Minecraft.getMinecraft().player, "item.supplyCampDeployer.invalid");
+                final Map<PlacementErrorType, List<BlockPos>> blockPosListByErrorTypeMap = PlacementError.partitionPlacementErrorsByErrorType(
+                        placementErrorList);
+                for (final Map.Entry<PlacementErrorType, List<BlockPos>> entry : blockPosListByErrorTypeMap.entrySet()) {
+                    final PlacementErrorType placementErrorType = entry.getKey();
+                    final List<BlockPos> blockPosList = entry.getValue();
+
+                    final int numberOfBlocksTOReport = blockPosList.size() > 5 ? 5 : blockPosList.size();
+                    final List<BlockPos> blocksToReportList = blockPosList.subList(0, numberOfBlocksTOReport);
+                    String outputList = PlacementError.blockListToCommaSeparatedString(blocksToReportList);
+                    if (blockPosList.size() > numberOfBlocksTOReport) {
+                        outputList += "...";
+                    }
+                    String errorMessage;
+                    switch(placementErrorType) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)